### PR TITLE
Add Presto inverse_cauchy_cdf function

### PR DESCRIPTION
**Pull request by jobbine-joseph**
implementing inverse Cauchy cdf

### CODE REVIEW

**Comment by amitkdutta** [comment]
@jobbine-joseph Please do a re-base to fix the linux build. Otherwise looks good.

File: velox/docs/functions/presto/math.rst [hunk 1]
```diff
@@ -53,7 +53,11 @@ Mathematical Functions
 
 .. function:: from_base(string, radix) -> bigint
 
-    Returns the value of ``string`` interpreted as a base-``radix`` number. ``radix`` must be between 2 and 36.
+   Returns the value of ``string`` interpreted as a base-``radix`` number. ``radix`` must be between 2 and 36.
+
+.. function::   inverse_cauchy_cdf(median, scale, p) → double#
+   Compute the inverse of the Cauchy cdf with given parameters median and scale (gamma) for the probability p.
```
[the remark this answers is not in the document]
Comment (jobbine-joseph) [reply velox/docs/functions/presto/math.rst:60]: Hi @aditi-pandit , thank you for the review , addressed your comment

**Comment by jobbine-joseph** [comment]
> @jobbine-joseph Please do a re-base to fix the linux build. Otherwise looks good.
Hi @amitkdutta,
Thank you for the review , addressed your suggestion.

File: velox/docs/functions/presto/math.rst [hunk 2]
```diff
@@ -53,7 +53,12 @@ Mathematical Functions
 
 .. function:: from_base(string, radix) -> bigint
 
-    Returns the value of ``string`` interpreted as a base-``radix`` number. ``radix`` must be between 2 and 36.
+   Returns the value of ``string`` interpreted as a base-``radix`` number. ``radix`` must be between 2 and 36.
+
+.. function::   inverse_cauchy_cdf(median, scale, p) → double#
```
Comment (aditi-pandit) [inline velox/docs/functions/presto/math.rst:58]: This function needs to be documented with the rest of the inverse probability functions in the last section "Probability functions : inverse cdf" part of this file.

File: velox/functions/prestosql/Probability.h [hunk 3]
```diff
@@ -111,13 +113,44 @@ struct CauchyCDFFunction {
...
+      result = std::numeric_limits<double>::quiet_NaN();
+    } else if (median == kInf || median == kDoubleMax) {
+      result = -kInf;
+    } else if (scale == kInf) {
+      result = median;
+    } else if (probability == 1.0) {
+      result = kInf;
+    } else if (scale <= 0) {
+      result = std::numeric_limits<double>::quiet_NaN();
+      VELOX_USER_CHECK_GT(scale, 0, "scale must be greater than 0");
```
Comment (aditi-pandit) [inline velox/functions/prestosql/Probability.h:141]: If the scale <= 0 as checked by the if condition, then the CHECK is always going to be hit. This code is very confusing. The CHECK should be for the code in the else part of the condition if at all.

Please can you double check.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 4]
```diff
@@ -154,6 +154,26 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
       binomialCDF(-2, 0.5, -1), "numberOfTrials must be greater than 0");
 }
+TEST_F(ProbabilityTest, cauchyInverseCDF) {
+  const auto cauchyInvCDF = [&](std::optional<double> median,
+                                std::optional<double> scale,
+                                std::optional<double> probability) {
+    return evaluateOnce<double>(
+        "inverse_cauchy_cdf(c0, c1, c2)", median, scale, probability);
+  };
+  EXPECT_DOUBLE_EQ(0.0, cauchyInvCDF(0.0, 1.0, 0.5).value());
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:164]: Does EXPECT_EQ not work for these ?

Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:164]: Please add test cases with std::nulllopt

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 5]
```diff
@@ -154,6 +154,26 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
+  EXPECT_DOUBLE_EQ(1.0, cauchyInvCDF(0.0, 1.0, 0.75).value());
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:165]: Please add test cases with kDoubleMax, kDoubleMin, kInf, kNan for each parameter.

**Comment by yzhang1991** [comment]
Hi @jobbine-joseph, just in case you are not aware, we are holding on merging any inverse function implementations for now due to this issue https://github.com/facebookincubator/velox/pull/5382#issuecomment-1651920130. We are waiting for a fix from Meta.
CC @aditi-pandit

**Pull request closed without merging**